File: mcp_processor/processor.py
import time
import random
from concurrent.futures import ThreadPoolExecutor, as_completed

def process_channel(channel_id, data):
    print(f'Channel {channel_id} start processing {data}')
    # 模拟处理耗时
    time.sleep(random.uniform(0.5, 2.0))
    result = f'Channel {channel_id} processed {data.upper()}'
    print(f'Channel {channel_id} done')
    return result

if __name__ == '__main__':
    channel_data = {
        1: 'alpha',
        2: 'bravo',
        3: 'charlie',
        4: 'delta',
    }
    results = []
    with ThreadPoolExecutor(max_workers=4) as executor:
        future_to_channel = {
            executor.submit(process_channel, cid, data): cid
            for cid, data in channel_data.items()
        }
        for future in as_completed(future_to_channel):
            cid = future_to_channel[future]
            try:
                res = future.result()
                results.append(res)
            except Exception as exc:
                print(f'Channel {cid} generated an exception: {exc}')
    print('\nAll results:')
    for r in results:
        print(r) 

File: mcp_processor_go/processor.go
package main

import (
	"fmt"
	"strings"
	"sync"
	"time"
)

func processChannel(id int, data string, wg *sync.WaitGroup, results chan<- string) {
	defer wg.Done()
	fmt.Printf("Channel %d start processing %s\n", id, data)
	time.Sleep(time.Duration(500+id*200) * time.Millisecond) // 模拟耗时
	result := fmt.Sprintf("Channel %d processed %s", id, strings.ToUpper(data))
	fmt.Printf("Channel %d done\n", id)
	results <- result
}

func main() {
	channelData := map[int]string{
		1: "alpha",
		2: "bravo",
		3: "charlie",
		4: "delta",
	}
	var wg sync.WaitGroup
	results := make(chan string, len(channelData))
	for id, data := range channelData {
		wg.Add(1)
		go processChannel(id, data, &wg, results)
	}
	wg.Wait()
	close(results)
	fmt.Println("\nAll results:")
	for r := range results {
		fmt.Println(r)
	}
}
